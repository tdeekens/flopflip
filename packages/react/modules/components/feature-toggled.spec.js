import React from 'react';
import { shallow } from 'enzyme';
import FeatureToggled from './feature-toggled';

const UntoggledComponent = <div>{'UntoggledComponent'}</div>;
const FeatureComponent = () => <div>{'FeatureComponent'}</div>;

describe('with feature disabled', () => {
  describe('with untoggled component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <FeatureToggled
          isFeatureEnabled={false}
          untoggledComponent={UntoggledComponent}
        >
          <FeatureComponent />
        </FeatureToggled>
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render the `UntoggledComponent`', () => {
      expect(wrapper).toHaveText('UntoggledComponent');
    });

    it('should not render the `FeatureComponent`', () => {
      expect(wrapper).not.toRender('FeatureComponent');
    });
  });

  describe('without untoggled component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <FeatureToggled
          isFeatureEnabled
          untoggledComponent={UntoggledComponent}
        >
          <FeatureComponent />
        </FeatureToggled>
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render the `UntoggledComponent`', () => {
      expect(wrapper).not.toRender('UntoggledComponent');
    });

    it('should render the `FeatureComponent`', () => {
      expect(wrapper).toRender('FeatureComponent');
    });
  });
});
