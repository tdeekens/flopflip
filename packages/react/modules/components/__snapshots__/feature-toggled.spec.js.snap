// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when feature disabled with untoggled component should match snapshot 1`] = `<UntoggledComponent />`;

exports[`when feature disabled without untoggled component should match snapshot 1`] = `""`;

exports[`when feature enabled with \`children\` should match snapshot 1`] = `<FeatureComponent />`;

exports[`when feature enabled with \`toggledComponent\` should match snapshot 1`] = `<FeatureComponent />`;
